### PR TITLE
Sanitise API inputs used as file path variables 

### PR DESCRIPTION
**Pull request by saurabhd336**
Adds sanity checks whenever API params are directly used to construct file paths

### CODE REVIEW

**Comment by saurabhd336** [comment]
cc: @soumitra-st

**Review by soumitra-st — APPROVED** [review]
LGTM

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java [hunk 1]
```diff
@@ -448,8 +448,13 @@ private static File extractSegmentFromFormToLocalTempFile(FormDataMultiPart form
...
       FormDataBodyPart bodyPart = map.values().iterator().next().get(0);
 
-      File localTempFile = new File(ControllerFilePathProvider.getInstance().getFileUploadTempDir(),
-          getTempSegmentFileName(segmentName));
+      File fileUploadTempDir = ControllerFilePathProvider.getInstance().getFileUploadTempDir();
+      File localTempFile = new File(fileUploadTempDir, getTempSegmentFileName(segmentName));
+
+      if (!localTempFile.getCanonicalPath().startsWith(fileUploadTempDir.getPath())) {
+        throw new IllegalArgumentException("Invalid segment name: " + segmentName);
+      }
```
Comment (walterddr) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java:456]: create a file path util (put in pinot-common/src/main/java/org/apache/pinot/common/utils/FileUtils.java)
```suggestion
public static File concatAndSanitizedFile(File folderDir, File filename) {
      File filePath = new File(folderDir, filename);

      if (!filePath.getCanonicalPath().startsWith(folderDir.getPath())) {
        throw new IllegalArgumentException("Invalid file name: " + filename);
      }
    }
```
  and then the util can be used across all the following

**Review by walterddr — APPROVED** [review]
lgtm mostly. minor nit

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java (hunk 1 above)
Comment (saurabhd336) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java]: Makes sense. Addressed

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java [hunk 2]
```diff
@@ -448,8 +448,10 @@ private static File extractSegmentFromFormToLocalTempFile(FormDataMultiPart form
           "Invalid multi-part for segment: %s", segmentName);
       FormDataBodyPart bodyPart = map.values().iterator().next().get(0);
 
-      File localTempFile = new File(ControllerFilePathProvider.getInstance().getFileUploadTempDir(),
-          getTempSegmentFileName(segmentName));
+      File localTempFile = org.apache.pinot.common.utils.FileUtils.concatAndValidateFile(
```
Comment (soumitra-st) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java:451]: Shall we import org.apache.pinot.common.utils.FileUtils?

**Review by soumitra-st — COMMENTED** [review]
Let's import org.apache.pinot.common.utils.FileUtils in all the files. Great work!

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java (hunk 2 above)
Comment (walterddr) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java:451]: unfortunately this class already imported common-io FileUtils. it has to use fully qualified package name

**Review by walterddr — APPROVED** [review]

**Pull request merged**